lib/tasks/plugin.rake

```diff
@@ -24,16 +28,26 @@ task "plugin:install_all_official" do
       repo += ".git"
     end
 
-    attempts = 0
-    begin
-      attempts += 1
-      system("git clone #{repo} #{path}", exception: true)
-    rescue StandardError
-      abort("Failed to clone #{repo}") if attempts >= 3
-      STDERR.puts "Failed to clone #{repo}... trying again..."
-      retry
+    Concurrent::Promise.execute do
+      attempts = 0
+      begin
+        attempts += 1
+        STDOUT.puts("Cloning '#{repo}' to '#{path}'...")
+        system("git clone --quiet #{repo} #{path}", exception: true)
+      rescue StandardError
+        if attempts >= 3
+          failures << repo
+          abort
+        end
+
+        STDOUT.puts "Failed to clone #{repo}... trying again..."
+        retry
+      end
     end
   end
+
+  Concurrent::Promise.zip(*promises).value!
```

Comment:
Is there a limit to the concurrency? Or will it try to completely saturate the bandwidth and the number of connections to a single host?

Reply:
It will spin X number of threads where X is equal to the number of CPU cores on the host. We could set an upper limit here but I doubt anyone will be running this one a CPU with enough cores to cause problems.

Reply:
Ah, nice!

Reply:
BTW, our internal CI host has 32 cores. Maybe still fine, but worth bearing in mind 👀

Reply:
O I have 32 cores locally too and nothing broke. I’ll be alittle aggressive for now with the number of concurrent git clones but will scale it back down should we run into problems.